src/vs/workbench/api/common/extHostNotebookKernels.ts

```diff
@@ -44,6 +44,7 @@ export class ExtHostNotebookKernels implements ExtHostNotebookKernelsShape {
 
 	private readonly _proxy: MainThreadNotebookKernelsShape;
 	private readonly _activeExecutions = new ResourceMap<NotebookCellExecutionTask>();
+	private readonly _activeNotebookExecutions = new ResourceMap<[NotebookExecutionTask, IDisposable]>();
```

Comment:
If the only reason this exists is to dispose these things (and track that an execution is in progress), then just make it a `ResourceMap<IDisposable>`

Reply:
Its also used to check if we have an existing NotebookExecution